### PR TITLE
powershell_script move validation into one shell_out!

### PR DESCRIPTION
**Pull request by smurawski**
@jaym This reverts your change from https://github.com/chef/chef/pull/3815, but reduces the number of shell_outs to one, where we get validation and execution from one powershell process.  /cc @chef/client-windows

### CODE REVIEW

File: lib/chef/provider/powershell_script.rb [hunk 1]
```diff
@@ -29,20 +29,6 @@ def initialize (new_resource, run_context)
         add_exit_status_wrapper
       end
 
-      def action_run
-        validate_script_syntax!
-        super
-      end
-
-      def command
```
Comment (ksubrama) [inline lib/chef/provider/powershell_script.rb]: Why did you remote command?  I don't think windows_script sets the interpreter path correctly.

Comment (smurawski) [inline lib/chef/provider/powershell_script.rb]: It's already in the resource and windows_script handles dealing with whether or not to use file redirection paths.  it only exists here for the validate step, which I moved into the wrapper script so we only have on shell out and don't have to deal with re-implementing the "should I use file system bitness redirection or not".

**Comment by smurawski** [comment]
Got a failing functional test for the resource.. gonna fix that one

**Comment by smurawski** [comment]
The travis failure appears unrelated to any changes in this PR.

**Comment by smurawski** [comment]
And looks like my build failures match the state of master.

File: lib/chef/provider/powershell_script.rb (hunk 1 above)
Comment (ksubrama) [inline lib/chef/provider/powershell_script.rb]: It actually doesn't do it fully.  It only works when 64-bit is requested on 32-bit ruby.  It doesn't handle 32-bit request on 64-bit.  This case was done by the line below that @chefsalim had added.  I had initially worried about consolidating this logic in one place and keeping with_os_architecture consistent for this reason but the refactor was non-trivial at the time.  Currently, with_os_architecture only does 32->64 fixes.  The reverse doesn't work similarly.

**Comment by lamont-granquist** [comment]
@chef/client-windows need a :+1: or two because this is over my head

File: lib/chef/provider/powershell_script.rb (hunk 1 above)
Comment (smurawski) [inline lib/chef/provider/powershell_script.rb]: @ksubrama That's all well and good, but powershell_script doesn't care about that either.. it delegates that decision to windows_script.  If we fix that logic, it should be there.

**Comment by btm** [comment]
Current output with a bonus quote:
```
[2015-11-23T18:51:36-05:00] FATAL: Mixlib::ShellOut::ShellCommandFailed: powershell_script[write-to-interpolated-path] ((chef-apply cookbook)::(chef-apply recipe) line 1) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '1'
---- Begin output of "powershell.exe" -NoLogo -NonInteractive -NoProfile -ExecutionPolicy Bypass -InputFormat None -Command ". 'C:/Users/btm_000/AppData/Local/Temp/chef_powershell_script-user-code20151123-24316-1ugk7hk.ps1'" ----
STDOUT:
STDERR: At C:\Users\btm_000\AppData\Local\Temp\chef_powershell_script-user-code20151123-24316-1ugk7hk.ps1:4 char:25
+     $stream.WriteLine(''foo')
+                         ~
Missing ')' in method call.
At C:\Users\btm_000\AppData\Local\Temp\chef_powershell_script-user-code20151123-24316-1ugk7hk.ps1:4 char:28
+     $stream.WriteLine(''foo')
+                            ~~
The string is missing the terminator: '.
At C:\Users\btm_000\AppData\Local\Temp\chef_powershell_script-user-code20151123-24316-1ugk7hk.ps1:4 char:25
+     $stream.WriteLine(''foo')
+                         ~~~~~
Unexpected token 'foo')
    $stream.close()
}
' in expression or statement.
At C:\Users\btm_000\AppData\Local\Temp\chef_powershell_script-user-code20151123-24316-1ugk7hk.ps1:1 char:1
+ {
+ ~
Missing closing '}' in statement block or type definition.
    + CategoryInfo          : ParserError: (:) [], ParseException
    + FullyQualifiedErrorId : MissingEndParenthesisInMethodCall
---- End output of "powershell.exe" -NoLogo -NonInteractive -NoProfile -ExecutionPolicy Bypass -InputFormat None -Command ". 'C:/Users/btm_000/AppData/Local/Temp/chef_powershell_script-user-code20151123-24316-1ugk7hk.ps1'" ----
Ran "powershell.exe" -NoLogo -NonInteractive -NoProfile -ExecutionPolicy Bypass -InputFormat None -Command ". 'C:/Users/btm_000/AppData/Local/Temp/chef_powershell_script-user-code20151123-24316-1ugk7hk.ps1'" returned 1
```

Output under this PR:
```
[2015-11-23T18:48:10-05:00] FATAL: Mixlib::ShellOut::ShellCommandFailed: powershell_script[write-to-interpolated-path] ((chef-apply cookbook)::(chef-apply recipe) line 1) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '5'
---- Begin output of "powershell.exe" -NoLogo -NonInteractive -NoProfile -ExecutionPolicy Bypass -InputFormat None -File "C:/Users/btm_000/AppData/Local/Temp/chef-script20151123-3892-1yn6uqi.ps1" ----
STDOUT:
STDERR: New-UserScriptBlock : Exception calling "NewScriptBlock" with "1" argument(s): "At line:3 char:25
+     $stream.WriteLine(''foo')
+                         ~
Missing ')' in method call.
At line:3 char:25
+     $stream.WriteLine(''foo')
+                         ~~~~~
Unexpected token 'foo')
    $stream.close()
  # This assignment doesn't' in expression or statement.
At line:7 char:45
+   # This assignment doesn't affect the block's return value
+                                             ~~~~~~~~~~~~~~~
The string is missing the terminator: '."
At C:\Users\btm_000\AppData\Local\Temp\chef-script20151123-3892-1yn6uqi.ps1:38 char:22
+ $chefscriptresult = (New-UserScriptBlock).invokereturnasis()
+                      ~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,New-UserScriptBlock
---- End output of "powershell.exe" -NoLogo -NonInteractive -NoProfile -ExecutionPolicy Bypass -InputFormat None -File "C:/Users/btm_000/AppData/Local/Temp/chef-script20151123-3892-1yn6uqi.ps1" ----
Ran "powershell.exe" -NoLogo -NonInteractive -NoProfile -ExecutionPolicy Bypass -InputFormat None -File "C:/Users/btm_000/AppData/Local/Temp/chef-script20151123-3892-1yn6uqi.ps1" returned 5
```

Looks pretty similar. There's some added noise from the wrapper which could be confusing, but it is all pretty noisy so I'm not that concerned. The return code of 5 is interesting.

**Pull request closed without merging**